chain-api/src/types/dtos.ts

```diff
@@ -15,6 +15,7 @@
 import { Type, instanceToInstance, plainToInstance } from "class-transformer";
 import { IsNotEmpty, IsOptional, ValidationError, validate } from "class-validator";
 import { JSONSchema } from "class-validator-jsonschema";
+import crypto from "crypto";
```

Comment:
Ref:
[Node.js v18.x crypto](https://nodejs.org/docs/latest-v18.x/api/crypto.html#determining-if-crypto-support-is-unavailable)
[Node.js v20.x crypto](https://nodejs.org/docs/latest-v20.x/api/crypto.html#determining-if-crypto-support-is-unavailable)

Especially since this is in `chain-api`, the most re-useable and general purpose module, It may be worth checking for crypto support as detailed in the linked Node.js documentation: 

```suggestion
let crypto;
try {
  crypto = await import('node:crypto');
} catch (err) {
  console.error('crypto support is disabled!'); 
} 
```

(maybe re-word or re-work the error / log handling)